www/src/app/shared/course-ability.service.ts

```diff
@@ -0,0 +1,49 @@
+import {Injectable} from '@angular/core';
+import {Observable} from 'rxjs/Observable';
+import {Subject} from 'rxjs/Subject';
+import {AppConstants} from "../app.constants";
+import {Course} from '../models/course.model';
+import {Principal} from "./auth/principal.service";
+
+@Injectable()
+export class CourseAbility {
+  constructor(private principal: Principal) {
+  }
+
+  canAddStudents(course: Course): boolean {
+    return this.hasCreateUpdateDeleteAuthority(course);
+  }
+
+  canEditGrade(course: Course): boolean {
+    return this.hasCreateUpdateDeleteAuthority(course);
+  }
+
+  canDelete(course: Course): boolean {
+    return this.hasCreateUpdateDeleteAuthority(course);
+  }
+
+  private hasCreateUpdateDeleteAuthority(course: Course): boolean {
+    return !this.isCurrentUserAStudent() &&
```

Comment:
I include that one because its one of the least expensive to check. That way if the current student is a student, it'll bail out with having to check 3 conditions every time. It's a small optimization, and doesn't probably have too large an impact, so if you'd prefer, I can remove it.